src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs

```diff
@@ -108,55 +107,75 @@ public HostingLogScope(HttpContext httpContext)
                 _httpContext = httpContext;
             }
 
-            public override string ToString() => _cachedToString ?? Interlocked.CompareExchange(
-                ref _cachedToString,
-                $"RequestId:{_httpContext.TraceIdentifier} RequestPath:{_httpContext.Request.Path}",
-                null);
+            public override string ToString()
+            {
+                if (_cachedToString == null)
+                {
+                    _cachedToString = $"RequestId:{_httpContext.TraceIdentifier} RequestPath:{_httpContext.Request.Path}";
+                }
+
+                return _cachedToString;
+            }
 
-            public IEnumerable<KeyValuePair<string, object>> GetValues() => _cachedGetValues ?? Interlocked.CompareExchange(
-                ref _cachedGetValues,
-                new[]
+            public IEnumerable<KeyValuePair<string, object>> GetValues()
+            {
+                if (_cachedGetValues == null)
                 {
-                    new KeyValuePair<string, object>("RequestId", _httpContext.TraceIdentifier),
-                    new KeyValuePair<string, object>("RequestPath", _httpContext.Request.Path.ToString()),
-                },
-                null);
+                    _cachedGetValues = new[]
+                    {
+                        new KeyValuePair<string, object>("RequestId", _httpContext.TraceIdentifier),
+                        new KeyValuePair<string, object>("RequestPath", _httpContext.Request.Path.ToString()),
+                    };
+                }
+
+                return _cachedGetValues;
+            }
         }
 
         private class HostingRequestStarting : ILogValues
         {
             internal static readonly Func<object, Exception, string> Callback = (state, exception) => ((HostingRequestStarting)state).ToString();
 
-            private readonly HttpContext _httpContext;
+            private readonly HttpRequest _request;
```

Comment:
This is nice, but why only in request starting? For example, we can do something similar for `_httpContext.Response` in `HostingRequestFinished`.